### ISSUE TITLE
Generating a report out of an XML that has 0 test results throws a Null Pointer exception

### ISSUE THREAD

**Issue by onurgokkocabas** [web-innovate/bootstraped-multi-test-results-report#74]
```
19:12:30 [RSpecReportPublisher] searching for files ...
19:12:30 [RSpecReportPublisher] Compiling RSpec Html Reports ...
19:12:30 [RSpec test report builder] Copying XML files from: /var/lib/jenkins/workspace/-enable-rspec-tests-for-all-ZAUBXYGJI74CC4YHRMYSI4GQUJ4FR2BFVL4O5L4GAA32ANOTJIFA to reports directory: /var/lib/jenkins/jobs/CSLREW/jobs/blueprints/branches/CSLR-431-ena.darvcs.ests-for-all/builds/4/rspec-reports-with-handlebars
19:12:30 [RSpecReportPublisher] Found 1 xml files.
19:12:30 [RSpec test report builder] 0. Found a xml file: postfix_rspec-results.xml
19:12:30 [RSpec test report builder] Generating HTML reports
19:12:30 processing: /var/lib/jenkins/jobs/CSLREW/jobs/blueprints/branches/CSLR-431-ena.darvcs.ests-for-all/builds/4/rspec-reports-with-handlebars/xmlData/postfix_rspec-results.xml
19:12:31 [RSpec test report builder] there was an error generating the reports: java.lang.NullPointerException
19:12:31 com.github.bogdanlivadariu.reporting.rspec.xml.models.TestSuitesModel.postProcess(TestSuitesModel.java:85)
19:12:31 com.github.bogdanlivadariu.reporting.rspec.builder.RSpecReportBuilder.<init>(RSpecReportBuilder.java:50)
19:12:31 com.github.bogdanlivadariu.jenkins.reporting.rspec.RSpecTestReportPublisher.generateReports(RSpecTestReportPublisher.java:143)
19:12:31 com.github.bogdanlivadariu.jenkins.reporting.rspec.RSpecTestReportPublisher.perform(RSpecTestReportPublisher.java:203)
19:12:31 org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:78)
19:12:31 org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:65)
19:12:31 org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:49)
19:12:31 hudson.security.ACL.impersonate(ACL.java:260)
19:12:31 org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:46)
19:12:31 java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
19:12:31 java.util.concurrent.FutureTask.run(FutureTask.java:266)
19:12:31 java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
19:12:31 java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
19:12:31 java.lang.Thread.run(Thread.java:745)
```

**Comment by BogdanLivadariu** [comment]
@onurgokkocabas can you attach the xml file/files ?

**Comment by onurgokkocabas** [comment]
NOTE:  Github does not support XML files as attachments, so I changed the extension to txt, change it back to xml please.

[postfix_rspec-results.txt](https://github.com/web-innovate/bootstraped-multi-test-results-report/files/781390/postfix_rspec-results.txt)

library used to convert to junit format was 'yarjuf'

**Comment by BogdanLivadariu** [comment]
@onurgokkocabas if your test results are converted to junit, then you can use the junit reporter for them.

from how the results you uploaded look like, have you altered them in any way ? as it does not have any test there, is just a blank testsuite.


If you can share with me the rspec initial report files that would be awesome ... as I've never got my hands on one in a while 
if you don't want to put it here you can email me `bogdan.livadariu@gmail.com`

**Comment by onurgokkocabas** [comment]
no, its ok.  that's exactly what it is though.  Empty report throws an exception during parsing. 
[x.txt](https://github.com/web-innovate/bootstraped-multi-test-results-report/files/783368/x.txt)

**Comment by onurgokkocabas** [comment]
Similar odd issue with an empty test suite from JUnit Report too:
```
14:15:47 [JUnitReportPublisher] Compiling JUnit Html Reports ...
14:15:47 [JUnit test report builder] Copying XML files from: /var/lib/jenkins/workspace/-enable-rspec-tests-for-all-ZAUBXYGJI74CC4YHRMYSI4GQUJ4FR2BFVL4O5L4GAA32ANOTJIFA to reports directory: /var/lib/jenkins/jobs/CSLREW/jobs/blueprints/branches/CSLR-431-ena.darvcs.ests-for-all/builds/8/junit-reports-with-handlebars
14:15:47 [JUnitReportPublisher] Found 1 xml files.
14:15:47 [JUnit test report builder] 0. Found a xml file: postfix_rspec-results.xml
14:15:47 [JUnit test report builder] Generating HTML reports
14:15:47 processing: /var/lib/jenkins/jobs/CSLREW/jobs/blueprints/branches/CSLR-431-ena.darvcs.ests-for-all/builds/8/junit-reports-with-handlebars/xmlData/postfix_rspec-results.xml
14:15:47 [JUnit test report builder] there was an error generating the reports: javax.xml.bind.UnmarshalException
14:15:47  - with linked exception:
14:15:47 [org.xml.sax.SAXParseException; systemId: file:/var/lib/jenkins/jobs/CSLREW/jobs/blueprints/branches/CSLR-431-ena.darvcs.ests-for-all/builds/8/junit-reports-with-handlebars/xmlData/postfix_rspec-results.xml; lineNumber: 1; columnNumber: 1; Premature end of file.]
14:15:47 javax.xml.bind.helpers.AbstractUnmarshallerImpl.createUnmarshalException(AbstractUnmarshallerImpl.java:335)
14:15:47 com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.createUnmarshalException(UnmarshallerImpl.java:563)
14:15:47 com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:249)
14:15:47 com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:214)
14:15:47 javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:157)
14:15:47 javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:162)
14:15:47 javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:171)
14:15:47 javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:189)
14:15:47 com.github.bogdanlivadariu.reporting.junit.builder.JUnitReportBuilder.processXmlReports(JUnitReportBuilder.java:64)
14:15:47 com.github.bogdanlivadariu.reporting.junit.builder.JUnitReportBuilder.<init>(JUnitReportBuilder.java:42)
14:15:47 com.github.bogdanlivadariu.jenkins.reporting.junit.JUnitTestReportPublisher.generateReport(JUnitTestReportPublisher.java:145)
14:15:47 com.github.bogdanlivadariu.jenkins.reporting.junit.JUnitTestReportPublisher.perform(JUnitTestReportPublisher.java:203)
14:15:47 org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:78)
14:15:47 org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:65)
14:15:47 org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:49)
14:15:47 hudson.security.ACL.impersonate(ACL.java:260)
14:15:47 org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:46)
14:15:47 java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
14:15:47 java.util.concurrent.FutureTask.run(FutureTask.java:266)
14:15:47 java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
14:15:47 java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
14:15:47 java.lang.Thread.run(Thread.java:745)
```

**Comment by BogdanLivadariu** [comment]
@onurgokkocabas the issue you've reported will get fixed in https://github.com/web-innovate/bootstraped-multi-test-results-report/pull/76

and will be contained in the next release `2.0.2`

**Comment by onurgokkocabas** [comment]
@BogdanLivadariu thanks a lot !

**Comment by BogdanLivadariu** [comment]
@onurgokkocabas  I've just released `2.0.2`
you can wait for jenkins to pick it up in the auto update
or you can download it from [here](http://repo.jenkins-ci.org/public/com/github/bogdanlivadariu/bootstraped-multi-test-results-report/2.0.2/bootstraped-multi-test-results-report-2.0.2.hpi) and install it manually

after you upgrade, and if the issue is fixed please close the issue 

thanks.

p.s. looking forward for new issues that may arise :)
